### ISSUE TITLE
Automatic updates of the Spine libraries

### ISSUE THREAD

**Issue by armiol** [SpineEventEngine/base#535]
At the moment the interdependencies between Spine libraries form a dependency tree, with the `base` library being the root:

```
`base` <═╗
  ^ ^    ╠═ `time` <═╗
  ║ │    ║           ║
  ║ │    ╚═══════════╚═ `core-java` <═╗<╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╮
  ║ │                                 ║                  ╎
  ║ ├─────────────────────────────────╚═ `gcloud-java`   ╎
  ║ │                                 ║                  ╎
  ║ ├─────────────────────────────────╚═ `jdbc-storage`  ╎
  ║ │                                 ║                  ╎
  ║ ╰─────────────────────────────────╚═ `web` <╌╌╌╌╌╌╌╌╌┤
  ║                                                      ╎
  ╚══════════════════════════════════════════════════════╰ `bootstrap


 <═══ - both compile- and build-time dependencies
 <─── — build-time dependencies
 <╌╌╌ - run-time-only dependencies

 ```

As one of the libraries gets updated, the dependants should be updated as well. Clearly, new releases of `base` library trigger the updates all down the tree.

In the scope of this task, the updates in the chain `base` <═ `time` <═ `core-java` should be automated. 

**Propagating `base` updates**

As soon as the `base` is updated to the version `X.Y.Z` in its `master`, `time` and `core-java` repositories should get new pull requests.

* In each of these PRs the version of the `base` library should be updated to `X.Y.Z`. 

* The own version of `core-java` and `time` should be set to either `X.Y.Z` (to match the `base`) or to their own incremental versions — whichever is greater. 

* As long as `core-java` depends on `time` as well, `core-java` should refer the _new_ version of `time` library. In this way no manual modification of the PR will be needed, once `time` is merged and published.

**Use case 1:**

The `base` library is updated to `1.5.12`.
The `core-java` library has `1.5.7` and `time` has `1.5.9`.

(Version update script is launched)

The `time` receives a new PR; the dependency onto `base` is set to `1.5.12`. The version of `time` is set to `1.5.12` as well.
The `core-java` receives a new PR, the dependency onto `base` is set to `1.5.12`. The version of `core-java` is set to `1.5.12` as well.

**Use case 2:**

The `base` library is updated to `1.5.10`.
The `core-java` library has `1.5.14` and `time` has `1.5.4`.

(Version update script is launched)

The `time` receives a new PR; the dependency onto `base` is set to `1.5.12`. The version of `time` is set to `1.5.12` as well.
The `core-java` receives a new PR, the dependency onto `base` is set to `1.5.12`. The version of `core-java` is set to `1.5.15` — to ensure the version increment is made.

It will be really cool if once the `time` update is merged into `master`, the build of the PR in the `core-java` repository is automatically re-triggered. However, that may be a tricky feature to implement.

**Propagating `time` updates**

As soon as the `time` library gets a new version released to its `master`, the `core-java` should get a new pull request with an increment to the own `core-java` version and the new version of `time`.

This is a rare case, but still it has happened, so it should be automated.
